src/interfaces/noise_model.cpp

```diff
@@ -0,0 +1,19 @@
+#include "gl_depth_sim/interfaces/noise_model.h"
+
+std::vector<float> gl_depth_sim::noise(std::vector<float> distance)
+{
+  float segma;
+  std::random_device r;
+  std::seed_seq seed2{r(), r(), r(), r(), r(), r(), r(), r()};
+  std::mt19937 e2(seed2);
+
+  for (int i = 0 ; i < distance.size() ;++i){
```

Comment:
Sign/unsigned comparison.

`distances.size()` returns an unsigned type. When using STL containers like this, using `std::size_t` is usually good enough. E.g. `int` -> `std::size_t`.

Reply:
Alternatively you can re-write the entire loop using C++11 `range-for`. For example.
```c++
```